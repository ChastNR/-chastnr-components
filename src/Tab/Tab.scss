.tb__btn {
  align-items: stretch;
  background: #fff;
  border-radius: 8px;
  border: none;
  box-shadow: 0 4px 24px rgba(0, 0, 0, 0.12);
  color: rgba(0, 0, 0, 0.8);
  cursor: pointer;
  display: flex;
  flex-direction: column;
  height: auto;
  margin-bottom: 20px;
  margin: 0;
  outline: none;
  padding: 0;
  position: relative;
  text-align: left;
  text-decoration: none;
  transition-timing-function: ease;
  transition: transform 0.3s, box-shadow 0.3s;
  width: 100%;

  .tb__box {
    -ms-flex: 1 1 auto;
    align-items: center;
    border-radius: inherit;
    border: 1px solid transparent;
    box-sizing: border-box;
    display: flex;
    flex: 1;
    height: 100%;
    justify-content: space-between;
    outline: none;
    padding: 15px;
    width: 100%;

    .tb__txt {
      font-family: inherit;
      font-size: 15px;
      font-weight: 400;
      letter-spacing: 0;
      line-height: 24px;
      word-wrap: break-word;
    }

    .tb__cnt {
      flex-shrink: 0;
      height: 48px;
      margin-left: 20px;
      width: 48px;
    }
  }

  &:hover {
    box-shadow: 0 12px 36px rgba(0, 0, 0, 0.2);
    transform: translateY(-4px);
  }

  &:last-of-type {
    margin-bottom: unset;
  }
}
