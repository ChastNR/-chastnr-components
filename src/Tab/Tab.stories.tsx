/* eslint-disable import/no-extraneous-dependencies */
import { ComponentMeta } from '@storybook/react';
import { Tab } from './Tab';

const componentMeta: ComponentMeta<typeof Tab> = {
  title: 'Tab',
  component: Tab,
  parameters: {
    layout: 'fullscreen',
  },
};

export const DefaultTab = () => <Tab animate={false}>Test</Tab>;

export const ConfigurableTab = {
  args: {
    animate: true,
    children: 'Text',
  },
};

export default componentMeta;
