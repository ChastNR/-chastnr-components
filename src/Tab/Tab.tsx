import "./Tab.scss";

interface TabProps {
  content?: React.ReactNode;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
}

const Tab: React.FC<TabProps> = ({ content, children, onClick }) => (
  <button
    className="tb__btn"
    onClick={onClick}
    style={{ cursor: onClick ? "default" : "unset" }}
    type="button"
  >
    <span className="tb__box" tabIndex={-1}>
      <span className="tb__txt">{children}</span>
      {content && <span className="tb__cnt">{content}</span>}
    </span>
  </button>
);

export default Tab;
