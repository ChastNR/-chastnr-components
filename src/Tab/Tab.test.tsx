import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Tab from './Tab';

describe('<Tab/>', () => {
  it('should match snapshot', () => {
    const tab = shallow(<Tab />);
    expect(toJson(tab)).toMatchSnapshot();
  });

  it('should have content component', () => {
    const tab = shallow(<Tab content={<div>Test content</div>} />);
    const spanComponent = tab.find('.tb__cnt');

    expect(spanComponent).not.toBeNull();
  });

  it('style cursor should be cursor', () => {
    const tab = shallow(<Tab onClick={() => {}} />);
    const buttonElement = tab.find('.tb__btn');
    const buttonStyles = buttonElement.get(0).props.style;

    expect(buttonStyles).toHaveProperty('cursor', 'cursor');
  });
});
