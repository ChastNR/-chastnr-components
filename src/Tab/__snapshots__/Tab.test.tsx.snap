// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab/> should match snapshot 1`] = `
<button
  class="tb__btn animate"
  style="cursor: unset;"
  type="button"
>
  <span
    class="tb__box"
    tabindex="-1"
  >
    <span
      class="tb__txt"
    />
  </span>
</button>
`;
